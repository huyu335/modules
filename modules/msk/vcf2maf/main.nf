
process VCF2MAF {
    tag "$meta.id"
    label 'process_low' 

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-b6fc09bed47d0dc4d8384ce9e04af5806f2cc91b:305092c6f8420acd17377d2cc8b96e1c3ccb7d26-0':
        'biocontainers/mulled-v2-b6fc09bed47d0dc4d8384ce9e04af5806f2cc91b:305092c6f8420acd17377d2cc8b96e1c3ccb7d26-0' }"

    input:

    tuple val(meta), path(vcf), path(ref_fasta)
    
    output:
    
    tuple val(meta), path("*.maf"), emit: maf


    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
   
    """
    vcf2maf.pl $args \\
    --normal-id ${meta.control_id} \\
    $args2 --tumor-id \\
    ${meta.case_id} \\
    --vcf-normal-id ${meta.control_id} \\
    --vcf-tumor-id ${meta.case_id} \\
    --input-vcf ${vcf} \\
    --output-maf ${prefix}.maf \\
    --ref-fasta ${ref_fasta} \\
    --vep-path /usr/local/bin


    cat <<-END_VERSIONS > versions.yml
    "${task.process}": 
        vcf2maf: \$(echo \$(vcf2maf.pl --help) | echo 'vcf2maf version 1.6.21 VEP v105')
    END_VERSIONS
    """
    
    stub:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    echo "stub test" >> ${prefix}.maf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}": 
        vcf2maf: \$(echo \$(vcf2maf.pl --help) | echo 'vcf2maf version 1.6.21 VEP v105')
    END_VERSIONS
    """
}


//standard VCF2MAF args arguments: --buffer-size 5000 --maf-center mskcc.org --min-hom-vaf 0.7 --ncbi-build GRCh37 
//standard VCF2MAF args2 arguments: --retain-info set,TYPE,FAILURE_REASON,MUTECT 