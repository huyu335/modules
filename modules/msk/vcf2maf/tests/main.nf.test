// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test genomenexus
nextflow_process {

    name "Test Process vcf2maf"
    script "../main.nf"
    process "vcf2maf"

    tag "modules"
    tag "modules_nfcore"
    tag "vcf2maf"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("vcf2maf-test") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ case_id:'tumor', control_id:'normal' ], // meta map
                    file("/Users/ksivaprakasam/Downloads/C-2HXC96-P001-d01_cl_aln_srt_MD_IR_FX_BR__aln_srt_IR_FX-duplex.DONOR22-TP_cl_aln_srt_MD_IR_FX_BR__aln_srt_IR_FX-duplex.mutect.vcf", checkIfExists: true),
                    file("/Users/ksivaprakasam/Downloads/Homo_sapiens_assembly19.fasta", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("vcf2maf-test-stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ case_id:'tumor', control_id:'normal' ], // meta map
                    file("/Users/ksivaprakasam/Downloads/C-2HXC96-P001-d01_cl_aln_srt_MD_IR_FX_BR__aln_srt_IR_FX-duplex.DONOR22-TP_cl_aln_srt_MD_IR_FX_BR__aln_srt_IR_FX-duplex.mutect.vcf", checkIfExists: true),
                    file("/Users/ksivaprakasam/Downloads/Homo_sapiens_assembly19.fasta", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
