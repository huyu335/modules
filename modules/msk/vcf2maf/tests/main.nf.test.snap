{
    "normal_tumor_sample1 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "tumor.normal",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "tumor.normal.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "maf": [
                    [
                        {
                            "id": "tumor.normal",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "tumor.normal.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "timestamp": "2024-02-28T13:04:15.921179"
    },
    "normal_tumor_sample1": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "tumor.normal",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "tumor.normal.maf:md5,43f84d3fc1e864e9a0a192e2e2765a23"
                    ]
                ],
                "maf": [
                    [
                        {
                            "id": "tumor.normal",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "tumor.normal.maf:md5,43f84d3fc1e864e9a0a192e2e2765a23"
                    ]
                ]
            }
        ],
        "timestamp": "2024-02-28T13:05:35.448181"
    }
}